pkg/cmd/secrets.go

```diff
@@ -380,6 +408,89 @@ func downloadSecrets(cmd *cobra.Command, args []string) {
 	utils.Log(fmt.Sprintf("Downloaded secrets to %s", filePath))
 }
 
+func templateSecrets(cmd *cobra.Command, args []string) {
+	localConfig := configuration.LocalConfig(cmd)
+
+	utils.RequireValue("token", localConfig.Token.Value)
+
+	filePath, err := utils.GetFilePath(args[0])
+	if err != nil {
+		utils.HandleError(err, "Unable to parse template file path")
+	}
+
+	var file []byte
+	file, err = ioutil.ReadFile(filePath)
+	if err != nil {
+		utils.HandleError(err, "Unable to read template file")
+	}
+
+	var outputFile *os.File
+	var outputFilePath string
+	output := cmd.Flag("output").Value.String()
+	if len(output) != 0 {
+		outputFilePath, err = utils.GetFilePath(output)
+		if err != nil {
+			utils.HandleError(err, "Unable to parse output file path")
+		}
+		outputFile, err = os.Create(outputFilePath)
```

Comment:
Can we avoid creating the file until the secrets fetch has succeeded? In the event that secrets fetch fails, we'll have created a blank file.

Reply:
Great call 👍